### PR TITLE
Labels: Add search box to labels

### PR DESCRIPTION
**Pull request by gtk-grafana**
Fixes https://github.com/grafana/explore-logs/issues/452

Unified search code as much as possible, also added fuzzy searching to allow for typos and string insensitive matching

### CODE REVIEW

File: src/Components/ServiceScene/Breakdowns/BreakdownSearchScene.tsx [hunk 1]
```diff
@@ -0,0 +1,85 @@
...
+import { fuzzySearch } from '../../../services/search';
+
+export interface BreakdownSearchSceneState extends SceneObjectState {
+  valueFilter?: string;
+}
+
+export class BreakdownSearchScene extends SceneObjectBase<BreakdownSearchSceneState> {
+  constructor() {
+    super({
+      valueFilter: '',
```
Comment (matyax) [inline src/Components/ServiceScene/Breakdowns/BreakdownSearchScene.tsx:17]: Just `value` or `filter` should be ok 👍

File: src/Components/ServiceScene/Breakdowns/FieldsBreakdownScene.tsx [hunk 2]
```diff
@@ -264,40 +265,17 @@ export class FieldsBreakdownScene extends SceneObjectBase<FieldsBreakdownSceneSt
...
             {body instanceof LayoutSwitcher && <body.Selector model={body} />}
-            {!loading && value !== ALL_VARIABLE_VALUE && (
-              <SearchInput
-                value={valueFilter}
-                onChange={model.onValueFilterChange}
-                onClear={model.clearValueFilter}
-                placeholder="Search for value"
-              />
+            {!loading && value !== ALL_VARIABLE_VALUE && search instanceof BreakdownSearchScene && (
+              <search.Component model={search} />
```
Comment (matyax) [inline src/Components/ServiceScene/Breakdowns/FieldsBreakdownScene.tsx:278]: <3

File: src/Components/ServiceScene/Breakdowns/BreakdownSearchScene.tsx [hunk 3]
```diff
@@ -0,0 +1,85 @@
...
+      const body = this.parent.state.body;
+      body?.forEachChild((child) => {
+        if (child instanceof ByFrameRepeater && child.state.body.isActive) {
+          let haystack: string[] = [];
+
+          child.iterateFrames((frames, seriesIndex) => {
+            const labelValue = getLabelValue(frames[seriesIndex]);
+            haystack.push(labelValue);
+          });
+          fuzzySearch(haystack, filter, (data) => {
```
Comment (matyax) [inline src/Components/ServiceScene/Breakdowns/BreakdownSearchScene.tsx:54]: <3

**Review by matyax — APPROVED** [review]
Amazing!

File: src/Components/ServiceScene/Breakdowns/BreakdownSearchScene.tsx (hunk 1 above)
Comment (gtk-grafana) [reply src/Components/ServiceScene/Breakdowns/BreakdownSearchScene.tsx]: Fixed

**Pull request merged**